sklearn/impute/tests/test_impute.py

```diff
@@ -1001,6 +1002,40 @@ def test_iterative_imputer_skip_non_missing(skip_complete):
         assert_allclose(X_test_est[:, 0], [11, 7, 12], rtol=1e-4)
 
 
+@pytest.mark.parametrize(
+    "rs_imputer, rs_estimator, rs_expect",
+    [(None, None, None), (401, None, 401),
+     (np.random.RandomState(seed=402), None, "RandomState"),
+     (None, 403, 403), (404, 405, 405),
+     (np.random.RandomState(seed=406), 407, 407),
+     (None, "string", "string"), (408, "string", "string"),
+     (np.random.RandomState(seed=409), "string", "string")]
+)
+def test_iterative_imputer_set_estimator_random_state(
```

Comment:
I am finding it hard to read this test. I think naming and semantics of `rs_expect` could be improved at least.